client/src/routes/stats/+page.svelte

```diff
@@ -1 +1,144 @@
-<h2>Stats Page</h2>
+<script lang="ts">
+  import CircularProgress from "@smui/circular-progress";
+
+  import { authStore as user } from "$stores/auth";
+  import type { UserData } from "$lib/firebase/firestore-types/users";
+  import { doc, onSnapshot } from "firebase/firestore";
+  import { userCollection } from "$lib/firebase/firestore-collections";
+  import type { Unsubscribe } from "firebase/auth";
+  import { onDestroy } from "svelte";
+
+  // variables
+  let userInfo: UserData | undefined;
+  let unsubscribeUser: Unsubscribe | undefined;
+  let totalPlayed: number;
+  let totalWinRatio: number;
+  let catLosses: number;
+  let catfishLosses: number;
+  let catRatio: number;
+  let catfishRatio: number;
+
+  // reactive calls
+  $: if ($user !== null) {
+    unsubscribeUser = onSnapshot(doc(userCollection, $user.uid), (userDoc) => {
+      userInfo = userDoc.data();
+    });
+  }
+  $: if (userInfo !== undefined) {
+    totalPlayed = userInfo.playedAsCat + userInfo.playedAsCatfish;
+    catLosses = userInfo.playedAsCat - userInfo.catWins;
+    catfishLosses = userInfo.playedAsCatfish - userInfo.catfishWins;
+    catRatio = userInfo.catWins / catLosses;
+    catfishRatio = userInfo.catfishWins / catfishLosses;
+    totalWinRatio = (userInfo.catWins + userInfo.catfishWins) / (catLosses + catfishLosses);
+  }
+  // lifecyle methods
+  onDestroy(() => {
+    unsubscribeUser?.();
+  });
+</script>
+
+<main>
+  {#if $user == null}
+    <div class="spinner-wraper">
+      <CircularProgress indeterminate />
+    </div>
+  {:else if userInfo == undefined}
+    <h1>Sad Kitty... No Stats Here</h1>
+  {:else}
+    <h1>{userInfo.displayName} Stats</h1>
+    <div class="userInfo-container">
+      <div class="totals">
+        <div>
+          <h3>Total Games Played</h3>
+          <output>{totalPlayed}</output>
+        </div>
+        <div>
+          <h3>Total Win/Loss Ratio</h3>
+          <output>{isNaN(totalWinRatio) || !isFinite(totalWinRatio) ? "N/A" : totalWinRatio.toFixed(2)}</output>
+        </div>
+      </div>
+      <div>
+        <h3>Cat Wins</h3>
+        <output>{userInfo.catWins}</output>
+      </div>
+      <div>
+        <h3>Cat Losses</h3>
+        <output>{catLosses}</output>
+      </div>
+      <div>
+        <h3>Catfish Wins</h3>
+        <output>{userInfo.catfishWins}</output>
+      </div>
+      <div>
+        <h3>Catfish Losses</h3>
+        <output>{catfishLosses}</output>
+      </div>
+      <div>
+        <h3>Win/Loss Ratio as Cat</h3>
+        <output>{isNaN(catRatio) || !isFinite(catRatio) ? "N/A" : catRatio.toFixed(2)}</output>
+      </div>
+      <div>
+        <h3>Win/Loss Ratio as Catfish</h3>
+        <output>{isNaN(catfishRatio) || !isFinite(catfishRatio) ? "N/A" : catfishRatio.toFixed(2)}</output>
+      </div>
```

Comment:
the order feels wierd. what if it was cat on one row and catfish on the other row?